common/utils.js

```diff
@@ -20,8 +20,9 @@ export function round(number, roundTo) {
 }
 
 export function getTimeStr(now, offset=0){
-  let dayPrefix = ''
-  let hours = now.getHours() + offset
+  let dayPrefix = '';
+  now.setHours(now.getHours() + offset);
```

Comment:
Does this work? I'm not sure how `setHours()` behaves if given a non-integer. I think we need to change the minutes value as well

Reply:
The time is stored in `now`. So we add `offset` (3.5 hours in my case) to `now`. Then when we do `now.getHours()` and `now.getMinutes()` we have the correct time in the right timezone.

Reply:
Yes, `now` stores the time, but it doesn't look it gets updated correctly. At least that's what I see when I try it in node. It looks like the `.5` part of `offset` is ignored:

```javascript
> t
2020-02-20T13:45:00.000Z
> t.setHours(t.getHours() + 1.5)
1582209900000
> t.getHours() // should be 15
15
> t.getMinutes() // should also be 15
45
> t
2020-02-20T14:45:00.000Z // should be 2020-02-20T15:15:00.000Z
```

Maybe it behaves differently on fitbit though, have you tested it?

Reply:
How about using `setMinutes()` instead?

```javascript
now.setMinutes(now.getMinutes() + offset * 60)
```

Reply:
Oh wow, you are totally right! setHours() just floors the input. I must have been sleeping when I tested it. I'll update the merge request.

Reply:
It's updated now